packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx

```diff
@@ -182,40 +189,90 @@ const DialogueRecommendationRow = ({ rowProps, classes, showSuggestedTopics }: D
     skip: !currentUser
   });
 
-  if (!currentUser) return <></>;
+  const { loading: tagLoading, error: tagError, data: tagData } = useQuery(gql`
+    query UserTopTags($userId: String!) {
+      UserTopTags(userId: $userId) {
+        tag {
+          name
+          _id
+        }
+        commentCount
+      }
+    }
+  `, {
+    variables: { userId: targetUser._id },
+  });
+
+  const { loading: postsLoading, error: postsError, data: postsData } = useQuery(gql`
+    query UsersReadPostsOfTargetUser($userId: String!, $targetUserId: String!, $limit: Int) {
+      UsersReadPostsOfTargetUser(userId: $userId, targetUserId: $targetUserId, limit: $limit) {
+        _id
+        title
+      }
+    }
+  `, {
+    variables: { userId: currentUser?._id, targetUserId: targetUser._id, limit : 3 },
+  });
+
+  const topTags:[TagWithCommentCount] = tagData?.UserTopTags;
+  const readPosts:DbPost[] = postsData?.UsersReadPostsOfTargetUser
```

Comment:
these are not `DbPost`; that's a purely server-side and objects on the client will never actually fit that type